Adapt memory query to support current NEST release

jarsi opened the pull request:
The API for getting the memory of a NEST job changed. The try except block first tries the old API. If it fails, it instead uses the newer one.

diesmann commented:
shouldn't this be the other way around to express that the regular case is the more modern one and the exception the outdated code?

jarsi commented:
Agreed, this makes more sense. I swapped the order and the updated branch would be ready for merging. Could we reopen this PR?
Thanks for the review!

jarsi commented:
Apparently pull requests that are closed and merged cannot be reopened again. Thus I opened a new PR addressing the comments of @diesmann , please see #11.
